### PR TITLE
Detect deleted components in affected stacks

### PR DESCRIPTION
**Pull request by aknysh**
## what

- **Deleted Component Detection**: `atmos describe affected` now detects components and stacks that have been **deleted** in HEAD (current branch) compared to BASE (target branch)
- **New Output Fields**: Added `deleted` (boolean) and `deletion_type` (string: `component` or `stack`) fields to the affected output schema
- **New Affected Reasons**: Added `deleted` and `deleted.stack` as valid affected reasons
- **Integration Test Fixture**: Added comprehensive test fixture at `tests/fixtures/scenarios/atmos-describe-affected-deleted-detection/`
- **Bug Fix**: Fixed issue where stacks that only exist in BASE (not HEAD) were not being discovered during comparison - the code now re-scans BASE directory for stack config files instead of just converting HEAD paths
- **Documentation**: Updated `describe-affected.mdx` with new fields, reasons, and CI/CD workflow examples
- **GitHub Actions Guide**: Added new section in `affected-stacks.mdx` for destroy workflow integration
- **Blog Post**: Added changelog entry announcing the feature
- **Roadmap**: Updated CI/CD initiative with shipped milestone (progress: 80%)

## why

- **Resource Leaks**: Previously, when components or stacks were removed from configuration, CI/CD pipelines had no automated way to detect these deletions and trigger `terraform destroy`
- **Manual Destruction Required**: Users had to manually identify and destroy removed components, which was error-prone and could lead to orphaned cloud resources
- **Incomplete CI/CD**: Pipelines couldn't fully automate the infrastructure lifecycle - they could only handle creation and modification, not deletion
- **Audit Gaps**: No automated tracking of what infrastructure was removed from configuration

## How It Works

The algorithm now performs a **second pass** over BASE stacks to detect deletions:

1. For each stack in BASE:
   - If stack doesn't exist in HEAD → all components marked as `deleted.stack`
   - If stack exists in HEAD → check each component
     - If component doesn't exist in HEAD → marked as `deleted`

2. Abstract components (`metadata.type: abstract`) are **not** reported as deleted since they are blueprints and not provisioned

## CI/CD Integration

Users can now separate apply and destroy workflows:

```bash
# Get modified components (for apply)
atmos describe affected --query '[.[] | select(.deleted != true)]'

# Get deleted components (for destroy)
atmos describe affected --query '[.[] | select(.deleted == true)]'
```

## Test Coverage

- **8 unit tests** for deletion detection logic covering:
  - Component deleted from stack
  - Entire stack deleted
  - Abstract components not reported
  - Stack filter support
  - Multiple component types
  - Edge cases (no components section, no deletions)
  
- **2 integration tests** using new fixture:
  - `TestDescribeAffectedDeletedComponentDetection`
  - `TestDescribeAffectedDeletedComponentFiltering`

## references

- PRD: `docs/prd/describe-affected-deleted-detection.md`
- Related to CI/CD simplification initiative on roadmap

## Summary by CodeRabbit

* **New Features**
  * Detects deleted components and stacks in describe-affected; adds two output fields: deleted and deletion_type.

* **Documentation**
  * Added PRD, blog post, CLI docs and CI/CD examples describing deleted-detection and destruction workflows.

* **Tests**
  * Added unit/integration tests and fixtures covering deletion scenarios, filtering, and abstract-component behavior.

* **Chores**
  * Bumped toolchain/dependencies and refreshed NOTICE license entries.

* **Mocks**
  * Added telemetry mock method to support feature-flag checks.

* **Errors**
  * Introduced a new sentinel error for no stack manifests found.

### CODE REVIEW

File: website/blog/2026-02-08-describe-affected-deleted-detection.mdx [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+slug: describe-affected-deleted-detection
+title: 'New: Describe Affected Now Detects Deleted Components and Stacks'
```
Comment (osterman) [inline website/blog/2026-02-08-describe-affected-deleted-detection.mdx]:
```suggestion
title: 'Describe Affected Now Detects Deleted Components and Stacks'
```

File: website/src/data/roadmap.js [hunk 2]
```diff
@@ -336,13 +336,14 @@ export const roadmapConfig = {
...
+        { label: 'Deleted component detection for destroy workflows', status: 'shipped', quarter: 'q1-2026', pr: 2063, docs: '/cli/commands/describe/affected', changelog: 'describe-affected-deleted-detection', description: 'Automatically detect components and stacks that have been deleted in the current branch compared to the target branch. Enables CI/CD pipelines to trigger terraform destroy workflows for removed infrastructure.', category: 'featured', priority: 'high', benefits: 'CI/CD pipelines can now separate apply and destroy workflows automatically. No more orphaned cloud resources from removed components.' },
```
Comment (osterman) [inline website/src/data/roadmap.js]:
```suggestion
        { label: 'Detect deleted components in affected stacks', status: 'shipped', quarter: 'q1-2026', pr: 2063, docs: '/cli/commands/describe/affected', changelog: 'describe-affected-deleted-detection', description: 'Automatically detect components and stacks that have been deleted in the current branch compared to the target branch. Enables CI/CD pipelines to trigger terraform destroy workflows for removed infrastructure.', category: 'featured', priority: 'high', benefits: 'CI/CD pipelines can now separate apply and destroy workflows automatically. No more orphaned cloud resources from removed components.' },
```
